src/test/java/com/redhat/darcy/webdriver/WebDriverCookieManagerAddGetDeleteCookieTest.java

```diff
@@ -0,0 +1,106 @@
+
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import com.redhat.darcy.web.Cookie;
+import com.redhat.darcy.webdriver.internal.TargetedWebDriver;
+import com.redhat.darcy.webdriver.internal.WebDriverCookieManager;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+import org.junit.runners.Parameterized.Parameters;
+import org.junit.runners.Parameterized.Parameter;
+import org.openqa.selenium.WebDriver;
+
+import java.time.LocalDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+
+@RunWith(Parameterized.class)
+public class WebDriverCookieManagerAddGetDeleteCookieTest {
+
+    private static final Object[][] parameterizedChars = {
+            {new Cookie("chocolate", "chip")},
+            {new Cookie("chocolate", "chip", "")},
+            {new Cookie("chocolate", "chip", "", LocalDateTime.now().plusDays(1))},
+            {new Cookie("chocolate", "chip", "", null)},
+            {new Cookie("chocolate", "chip", null, "", LocalDateTime.now().plusDays(1))},
+            {new Cookie("chocolate", "chip", null, "", null, false)},
+            {new Cookie("chocolate", "chip", null, "", null, false, false)},
+    };
+
+    @Parameter
+    public Cookie cookie;
+
+    @Parameters
+    public static Collection<Object[]> data() {
+        return Arrays.asList(parameterizedChars);
+    }
+
+    @Test
+    public void shouldAddCookie() {
+        TargetedWebDriver mockDriver = mock(TargetedWebDriver.class);
+        WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+
+        WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+        cookieManager.add(cookie);
+
+        verify(mockOptions)
+                .addCookie(new org.openqa.selenium.Cookie(cookie.getName(), cookie.getValue()));
+    }
+
+    @Test
+    public void shouldGetCookie() {
+        TargetedWebDriver mockDriver = mock(TargetedWebDriver.class);
+        WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+
+        WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+        cookieManager.get(cookie);
+
+        verify(mockOptions)
+                .getCookieNamed(cookie.getName());
+    }
+
+    @Test
+    public void shouldDeleteCookie() {
+        TargetedWebDriver mockDriver = mock(TargetedWebDriver.class);
+        WebDriver.Options mockOptions = mock(WebDriver.Options.class);
+
+        when(mockDriver.manage())
+                .thenReturn(mockOptions);
+
+        WebDriverCookieManager cookieManager = new WebDriverCookieManager(mockDriver);
+        cookieManager.delete(cookie);
+
+        verify(mockOptions)
+                .deleteCookieNamed(cookie.getName());
+    }
```

Comment:
I don't think either of these tests need to be here, because they are testing the same thing either way I think (the only parameter used is the cookie's name regardless of what other state the cookie has).